model/cluster_utility.go

```diff
@@ -22,32 +25,91 @@ const (
 	TeleportCanonicalName = "teleport"
 )
 
-const (
+var (
 	// PrometheusOperatorDefaultVersion defines the default version for the Helm chart
-	PrometheusOperatorDefaultVersion = "9.4.4"
+	PrometheusOperatorDefaultVersion = &HelmUtilityVersion{Chart: "9.4.4", ValuesPath: "production"}
 	// ThanosDefaultVersion defines the default version for the Helm chart
-	ThanosDefaultVersion = "2.4.3"
+	ThanosDefaultVersion = &HelmUtilityVersion{Chart: "2.4.3", ValuesPath: "production"}
 	// NginxDefaultVersion defines the default version for the Helm chart
-	NginxDefaultVersion = "2.15.0"
+	NginxDefaultVersion = &HelmUtilityVersion{Chart: "2.15.0", ValuesPath: "production"}
 	// FluentbitDefaultVersion defines the default version for the Helm chart
-	FluentbitDefaultVersion = "2.8.7"
+	FluentbitDefaultVersion = &HelmUtilityVersion{Chart: "2.8.7", ValuesPath: "production"}
 	// TeleportDefaultVersion defines the default version for the Helm chart
-	TeleportDefaultVersion = "0.3.0"
+	TeleportDefaultVersion = &HelmUtilityVersion{Chart: "0.3.0", ValuesPath: "production"}
 )
 
+// UtilityVersion is an interface that provides the necessary methods
+// to discover the numerical version of the Utility as well as an
+// identifier necessary in order to fetch any other configuration
+type UtilityVersion interface {
```

Comment:
I am not sure if I fully understand the idea behind this interface. It seems like the only thing that implements it is `HelmUtilityVersion` and it introduces some complications like custom unmarshal (which is hard to debug after you forget about it).
Could you clarify it for me?

Reply:
Yeah, so, there's a somewhat historical reason for it. Despite the fact that all of the cluster utilities today are installed and managed using Helm, there has been a long standing desire to either replace Helm or at least be flexible in order to choose a different tool (or even replace these with operators). The idea for the interface is to allow that flexibility.

Now, that said, both you and @gabrieljackson have expressed some concern over this interface. If you guys feel strongly about it, I'm willing to simplify the implementation to use the concrete `HelmUtilityVersion` type throughout. The interface is here because I have been under the belief that we value the flexibility here.

Reply:
Thank you for the clarification!
I think I understand now. My intuition would be to keep it simple until we will have some more concrete idea of what and how we want to use instead of Helm, especially if it will not impact the API or data model.

Reply:
Yeah, I see the point that Szymon is raising here and I would say I am 1/5 on keeping it concrete/simple as well. I will leave it up to you to decide though Ian.

Reply:
It's fine if we don't want the interface here, but I will say, it's a surprise to me. This feature would've been a lot simpler to implement without it in the first place :joy:

Ah well, such is life. I'll simplify this to just use the concrete type. If we introduce something besides Helm for managing utilities, though, and need to reintroduce an interface here for that purpose, someone else is getting that task :rofl: